# Cài đặt
Import các thư viện liên quan chuẩn bị cho qua trình Thống kê Mô tả.
## Import thư viện

In [14]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import plotly.express as px

## Đọc dữ liệu

In [ ]:
df = pd.read_csv("winequality-red.csv")

## Hiểu dữ liệu


In [9]:
# Số hàng, số cột của dataset
print("(Số dòng, số cột) = " , str(df.shape))
# Hiển thị 5 hàng đầu tiên của dataset
df.head()

(Số dòng, số cột) =  (1599, 12)


fixed acidity  volatile acidity  citric acid  residual sugar  chlorides  \
0            7.4              0.70         0.00             1.9      0.076   
1            7.8              0.88         0.00             2.6      0.098   
2            7.8              0.76         0.04             2.3      0.092   
3           11.2              0.28         0.56             1.9      0.075   
4            7.4              0.70         0.00             1.9      0.076   

   free sulfur dioxide  total sulfur dioxide  density    pH  sulphates  \
0                 11.0                  34.0   0.9978  3.51       0.56   
1                 25.0                  67.0   0.9968  3.20       0.68   
2                 15.0                  54.0   0.9970  3.26       0.65   
3                 17.0                  60.0   0.9980  3.16       0.58   
4                 11.0                  34.0   0.9978  3.51       0.56   

   alcohol  quality  
0      9.4        5  
1      9.8        5  
2      9.8        5  
3      9.8        6  
4      9.4        5

Tổng cộng dataset có 1599 hàng và 12 cột. Nhìn có vẻ không có vấn đề nhưng cần kiểm tra thêm để chắc chắn không có hàng nào có giá trị bị thiếu.

## Giá trị bị thiếu

In [11]:
# Hiển thị giá trị bị thiếu cho từng cột
print(df.isna().sum())

fixed acidity           0
volatile acidity        0
citric acid             0
residual sugar          0
chlorides               0
free sulfur dioxide     0
total sulfur dioxide    0
density                 0
pH                      0
sulphates               0
alcohol                 0
quality                 0
dtype: int64


Dataset hiện tại khá thân thiện, không có cột nào bị thiếu giá trị. Vì thế không cần xử lý thêm các bước để xử lý mà có thể tiến hành vào phân tích, thống kê ngay.

## Khai phá biến dữ liệu

### Biểu đồ của biến "quality"


In [ ]:
fig = px.histogram(df,x='quality', 
                   labels={'quality': 'Chất lượng rượu', 'count': 'Số lượng'},
                   title="Biểu đồ phân phối chất lượng rượu")
fig
fig.show()